lib/constructor/index.js

```diff
@@ -292,103 +268,85 @@ Constructor.prototype.buildPerLocale = function buildPerLocale(opts, next) {
   // exist in an external service. We use 4 here so that the `finished`
   // event is the only 100 which gets sent when done is called
   //
-  progress.start(id, 4);
+  progress.start(id, 2);
   const current = assign({
-    locale: locale,
-    id: id
+    locale,
+    id
   }, omit(spec, 'locales'));
 
-  app.withBreadcrumb({
-    buildID: id,
+  app.contextLog.info('Start build for locale %s', locale, {
     locale: locale,
-    name: spec.name
-  }, app.contextLog.info, function runBuild() {
-    app.contextLog.info('Start build for locale %s', locale, {
-      locale: locale,
-      name: spec.name,
-      version: spec.version,
-      env: spec.env,
-      id: id
-    });
-    current.destDir = path.join(os.tmpdir(), `${current.id}-publish`);
-
-    /**
-     * Report the error to the developer before ending the async loop.
-     *
-     * @param {Error} err Failed step of the build process.
-     * @param {String} type Type of error
-     * @api private
-     */
-    function step(err, type) {
-      const key = constructor._key(current);
-      if (err) {
-        progress.fail(err, id);
-        app.contextLog.error('Error in step %s for %s: %s', type, spec.name, err.message, {
-          locale: locale,
-          name: spec.name,
-          version: spec.version,
-          env: spec.env,
-          id: id
-        });
-
-        constructor.failures[key] = constructor.failures[key] || 0;
-        if (++constructor.failures[key] >= constructor.maxFailures)
-          return next(err);
-      }
-      next();
-    }
+    name: spec.name,
+    version: spec.version,
+    env: spec.env,
+    id: id
+  });
 
-    //
-    // Launch the build process with the specifications and attach
-    // a supervisor to communicate all events back to the developer.
-    //
-    progress.write({
-      progress: true,
-      id: id,
-      message: `Running ${current.type} build ${current.name}`
-    });
+  /**
+    * Report the error to the developer before ending the async loop.
+    *
+    * @param {Error} err Failed step of the build process.
+    * @param {String} type Type of error
+    * @returns {undefined}
+    * @api private
+    */
+  function step(err, type) {
+    const key = constructor._key(spec);
+    if (err) {
+      progress.fail(err, id);
+      app.contextLog.error('Error in step %s for %s: %s', type, spec.name, err.message, {
```

Comment:
Would it make sense to mixin the error with the object so any added properties of the err are also logged. It could be a custom error with additional debug information?